src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -972,6 +973,400 @@ static void Main()
+                //         Unrelated([2]); // Span<string>
+                Diagnostic(ErrorCode.ERR_AmbigCall, "Unrelated").WithArguments("Program.Unrelated(System.Span<int>)", "Program.Unrelated(string[])").WithLocation(8, 9),
+                // 1.cs(9,9): error CS0121: The call is ambiguous between the following methods or properties: 'Program.Unrelated(Span<int>)' and 'Program.Unrelated(string[])'
+                //         Unrelated([string.Empty]); // string[]
+                Diagnostic(ErrorCode.ERR_AmbigCall, "Unrelated").WithArguments("Program.Unrelated(System.Span<int>)", "Program.Unrelated(string[])").WithLocation(9, 9));
+
+            string sourceB2 = """
+                partial class Program
+                {
+                    static void Main()
+                    {
+                        Unrelated(new[] { default }); // error
+                        Unrelated([default]); // ambiguous
+                    }
+                }
+                """;
+            comp = CreateCompilation(
+                new[] { sourceA, sourceB2 },
+                targetFramework: TargetFramework.Net80);
+            comp.VerifyEmitDiagnostics(
+                // 1.cs(5,19): error CS0826: No best type found for implicitly-typed array
+                //         Unrelated(new[] { default }); // error
+                Diagnostic(ErrorCode.ERR_ImplicitlyTypedArrayNoBestType, "new[] { default }").WithLocation(5, 19),
+                // 1.cs(5,19): error CS1503: Argument 1: cannot convert from '?[]' to 'System.Span<int>'
+                //         Unrelated(new[] { default }); // error
+                Diagnostic(ErrorCode.ERR_BadArgType, "new[] { default }").WithArguments("1", "?[]", "System.Span<int>").WithLocation(5, 19),
+                // 1.cs(6,9): error CS0121: The call is ambiguous between the following methods or properties: 'Program.Unrelated(Span<int>)' and 'Program.Unrelated(string[])'
+                //         Unrelated([default]); // ambiguous
+                Diagnostic(ErrorCode.ERR_AmbigCall, "Unrelated").WithArguments("Program.Unrelated(System.Span<int>)", "Program.Unrelated(string[])").WithLocation(6, 9));
+        }
+
+        [Fact]
+        public void BetterConversionFromExpression_04()
+        {
+            string source = """
+                using System;
+                class Program
+                {
+                    static void F1(int[] x, int[] y) { throw null; }
+                    static void F1(Span<object> x, ReadOnlySpan<int> y) { }
+                    static void F2(object x, string[] y) { throw null; }
+                    static void F2(string x, Span<object> y) { }
+                    static void Main()
+                    {
+                        F1([1], [2]);
+                        F2("3", ["4"]);
+                    }
+                }
+                """;
+            CompileAndVerify(
```

Comment:
Consider indicating which overloads were chosen

Reply:
Updated in https://github.com/dotnet/roslyn/pull/69627.